### PR TITLE
Improve Test Coverage

### PR DESCRIPTION
**Pull request by varshith257**
## Purpose

This PR aims to increase the test coverage of the StarbaseDB project to 75%+ using Vitest. The improvements ensure meaningful tests that not only improve line coverage but also help detect potential issues by testing different scenarios and edge cases and fixed few of them too

## Tasks

-  Added unit tests 
-  Fixed found issues too

- [ ]

## Verify

- Run the test suite using:
```
pnpm vitest run --coverage
```
- Confirm that all tests pass and coverage has increased

## After

/claim #71

### CODE REVIEW

**Comment by Brayden** [comment]
@varshith257 No worries. I was just updating your PR so it didn't have merge conflicts. I introduced a couple test changes so a report could be displayed on new PR's. I ran the test coverage here and figured I'd share:

image

Also do you recall what error you were getting for DELETE operations? Curious to what bug was uncovered!

**Comment by Brayden** [comment]
@varshith257 Wanted to check in and see if you had more ready to push up or not! :)

**Comment by varshith257** [comment]
@Brayden I will push the remainging changes by EOD. And maybe your eye on them would be good as some of tests are frequently failing as i am testing them wrong or maybe that tests we don't need

**Comment by Brayden** [comment]
> @Brayden I will push the remainging changes by EOD. And maybe your eye on them would be good as some of tests are frequently failing as i am testing them wrong or maybe that tests we don't need

Of course @varshith257, happy to take a look at the tests and verify them 👍

**Comment by Brayden** [comment]
> @Brayden This is the failing test I don't know where it going wrong here as id receiving as undefined after pasing request

I'll take a look at this!

**Comment by Brayden** [comment]
@varshith257 It looks like in the `./src/literest/index.test.ts` file all the tests are for the `external` source but we should probably make it for the `internal` source. For the error you're seeing, the paths should require `/rest/${schema}/${table}/${identifier}` but right now it looks like you're just doing `/rest/table/identifier` and the operations don't have access to a schema value.

We should probably do:
1. Make the testing source `internal` instead of `external`
2. Add in `main` as the schema value in the URL path for each REST operation

For the delete operation since the schema is missing and the last path value is the identifier it's acting as if no primary key identifier was passed into it so it doesn't know exactly what to delete.

File: src/rls/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                table: 'users',
+                column: 'user_id',
+                value: 'context.id()',
+                value_type: 'string',
+                operator: '=',
+            },
+        ])
+    })
+
+    it('should modify SELECT queries with WHERE conditions', async () => {
```
[the remark this answers is not in the document]
Comment (Brayden) [reply src/rls/index.test.ts:88]: Is this loading the policies like you do in the `loadPolicies` test? It should check the policies and if a policy exists for a particular table (in this case `users`) then it should be applied. I don't see the policies in play in this test but I could be missing it.

[the remark this answers is not in the document]
Comment (Brayden) [reply src/rls/index.test.ts:88]: Oh wait I see it in the `beforeEach` above this 🤦

[the remark this answers is not in the document]
Comment (Brayden) [reply src/rls/index.test.ts:88]: Looks like if I convert the `schema` from "public" to an empty string I get this result for this test:

```
 FAIL  src/rls/index.test.ts > applyRLS - Query Modification > should modify SELECT queries with WHERE conditions
AssertionError: expected 'SELECT * FROM `users` WHERE (`users`.…' to contain 'WHERE `user_id` = \'user123\''

Expected: "WHERE `user_id` = 'user123'"
Received: "SELECT * FROM `users` WHERE (`users`.`user_id` = 'user123')"

 ❯ src/rls/index.test.ts:114:29
    112|     it('should modify UPDATE queries with additional WHERE clause', async () => {
    113|         const sql = "UPDATE users SET name = 'Bob' WHERE age = 25"
    114|         const modifiedSql = await applyRLS({
       |                             ^
    115|             sql,
    116|             isEnabled: true,
    ```
    
    Which probably indicates an issue in the code that should handle default schemas.
